java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -454,38 +436,13 @@ class ReferenceModeStore private constructor(
     }
 
     /**
-     * The initialization of this property launches a flow that monitors the callback state (via
-     * the [callbacksStateChannel] and the [receiveQueue] size (via the [receiveQueue.sizeChannel].
-     * The first time all of these conditions are true:
-     * - callback count has transitioned to greater than 0 at least once
-     * - callback count is currently 0
-     * - [receiveQueue] size is empty
-     *
-     * then the backingStore entries will be cleaned up. The cleanup will occur on the
-     * [cleanupCoroutineContext] passed in the constructor. Currently, this is simply the
-     * [coroutineContext] used to construct the [ReferenceModeStore] via the [create] method.
-     */
-    @FlowPreview
-    private val backingStoreCleanupJob: Job = combine(
-            callbacksStateChannel.asFlow(),
-            receiveQueue.sizeChannel.asFlow()
-        ) { callbacksState, queueSize -> queueSize + if (callbacksState) 1 else 0 }
-            .filter { it == 0 }
-            .onEach {
-                if (receiveQueue.size.value == 0) {
-                    backingStore.clearStoresCache()
-                    receiveQueue.sizeChannel.close()
-                    callbacksStateChannel.close()
-                }
-            }
-            .launchIn(CoroutineScope(cleanupCoroutineContext + Job()))
-
-    /**
-     * This method will suspend until conditions lead to internal backing store cleanup to occur.
+     * This will be triggered whenever the callbacks list becomes empty, or the receive queue
```

Comment:
this comment is a bit confusing. The method will be called when the receive queue is drained. The cleaning will be done when the receiver queue is drained AND the callbacks list has become empty. Is that right?